3MI0800270/2023-02-28/zad1.cpp

```diff
@@ -0,0 +1,80 @@
+#include <iostream>
+using namespace std;
+
+struct transition {
+    size_t to;
+    char c;
+};
+
+struct state {
+    transition transitions[4];
+    size_t num;
+};
+
+struct stateInfo {
+    state data[50];
+};
+
+bool hasTransition(const stateInfo& arg, size_t from, char c) {
+    size_t number = arg.data[from].num;
+    for (int j = 0;j < number;j++) {
+        if (arg.data[from].transitions[j].c == c) {
+            return true;
+        }
+    }
+}
+bool addTransition(stateInfo& arg, size_t from, char c, size_t to) {
+    size_t number = arg.data[from].num;
+    if (number >= 4) {
+        return false;
+    }
+    else {
+        arg.data[from].transitions[number].c = c;
+        arg.data[from].transitions[number].to = to;
+    }
+}
+int getTransition(const stateInfo& arg, size_t from, char c) {
+    size_t number = arg.data[from].num;
+    for (int i = 0;i < number;i) {
+        if (arg.data[from].transitions[i].c == c) {
+            return i;
+        }
+    }
+    return -1;
+}
+bool canRead(const stateInfo& arg, char* word, size_t from) {
+    size_t pos = from;
+    while (!*word) {
+        if (hasTransition(arg, pos, *word)) {
+            pos = arg.data[pos].transitions[getTransition(arg, pos, *word)].to;
+            word++;
+        }
+        else {
+            break;
+        }
+    }
+    if (*word) {
```

Comment:
this whole if/else statement can be written as `return *word == '\0';` instead :))